File: app/src/main/java/com/past/music/widget/MineItemView.java
package com.past.music.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.past.music.pastmusic.R;

/**
 * =======================================================
 * 作者：gaojin
 * 日期：2017/1/27 下午8:28
 * 版本：
 * 描述：
 * 备注：
 * =======================================================
 */
public class MineItemView extends RelativeLayout {

    ImageView mImageView;
    TextView mItemName;
    TextView mItemCount;

    private int count;
    private String item_text;
    private Drawable icon;


    public MineItemView(Context context) {
        this(context, null);
    }

    public MineItemView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public MineItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        inial(context, attrs);
        initView();
    }

    private void inial(Context context, AttributeSet attrs) {
        TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.item_view);
        count = typedArray.getInt(R.styleable.item_view_count, 151);
        item_text = typedArray.getString(R.styleable.item_view_text);
        icon = typedArray.getDrawable(R.styleable.item_view_icon_view);
        typedArray.recycle();
    }

    private void initView() {
        View.inflate(getContext(), R.layout.view_mine_item_view, this);

        mImageView = findViewById(R.id.img_item);
        mItemName = findViewById(R.id.tv_item_name);
        mItemCount = findViewById(R.id.tv_item_count);

        mImageView.setImageDrawable(icon);
        mItemName.setText(item_text);
        mItemCount.setText(String.valueOf(count));
    }

    public void setmImageView(Drawable icon) {
        if (mImageView != null) {
            mImageView.setImageDrawable(icon);
        }
    }

    public void setmItemName(String item_text) {
        if (mItemName != null) {
            mItemName.setText(item_text);
        }
    }

    public void setmItemCount(String count) {
        if (mItemCount != null) {
            mItemCount.setText(count);
        }
    }
}

File: app/src/main/java/com/se/music/widget/MineItemView.kt
package com.se.music.widget

import android.content.Context
import android.graphics.drawable.Drawable
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.se.music.pastmusic.R

/**
 * Created by gaojin on 2018/2/26.
 */
class MineItemView : LinearLayout {

    private var mImageView: ImageView? = null
    private var mItemName: TextView? = null
    private var mItemCount: TextView? = null

    private var count: Int = 0
    private var itemtext: String? = null
    private var icon: Drawable? = null

    constructor(context: Context) : this(context, null)

    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(context, attrs, defStyleAttr, 0)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        init(context, attrs)
    }

    fun init(context: Context, attrs: AttributeSet?) {
        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.item_view)
        count = typedArray.getInt(R.styleable.item_view_count, 151)
        itemtext = typedArray.getString(R.styleable.item_view_text)
        icon = typedArray.getDrawable(R.styleable.item_view_icon_view)
        typedArray.recycle()

        View.inflate(getContext(), R.layout.view_mine_item_view, this)

        mImageView = findViewById(R.id.img_item)
        mItemName = findViewById(R.id.tv_item_name)
        mItemCount = findViewById(R.id.tv_item_count)

        mImageView!!.setImageDrawable(icon)
        mItemName!!.text = itemtext
        mItemCount!!.text = count.toString()
    }

    fun setmImageView(icon: Drawable) {
        if (mImageView != null) {
            mImageView!!.setImageDrawable(icon)
        }
    }

    fun setmItemName(item_text: String) {
        if (mItemName != null) {
            mItemName!!.text = item_text
        }
    }

    fun setmItemCount(count: String) {
        if (mItemCount != null) {
            mItemCount!!.text = count
        }
    }
}